esmvalcore/preprocessor/_area.py

```diff
@@ -475,18 +511,30 @@ def extract_shape(cube,
     See Also
     --------
     extract_region : Extract a region from a cube.
-
     """
-
     with fiona.open(shapefile) as geometries:
 
+        # get parameters specific to the shapefile (NE used case
+        # eg longitudes [-180, 180] or latitude missing
+        # or overflowing edges)
+        cmor_coords = True
+        pad_north_pole = False
+        pad_hawaii = False
+        if geometries.bounds[0] < 0:
+            cmor_coords = False
+        if geometries.bounds[1] > -90. and geometries.bounds[1] < -85.:
```

Comment:
@valeriupredoi 
I can reproduce the problem you mentioned with bands of NaN's at the edges of the NE shapefiles when I set pad_north_pole and pad_hawaii to be False and cmor_coords also to be True for the run (so your defaults in line 520-522). However the problem is solved already for me if I enable the cmor_coords check from line 523-524. Seems to me that the paddings are not needed.

Reply:
so that means `cmor_coords`, `pad_hawaii` and `pad_north_pole` are all `False`?

Reply:
Affirmative!

Reply:
> so that means cmor_coords, pad_hawaii and pad_north_pole are all False?

@valeriupredoi Did you check this? What do you think?

Reply:
looking at it now :beer:

Reply:
aha, took the fixes out the conditional if the coords are not cmor-like, it should be fixed now - the test is there to confirm :beer:

Reply:
Hi V,
Are you sure we cannot delete these padding fixes all together? My point was that the ne_files did look perfectly fine to me without the pad_hawaii and pad_north_pole fixes altogether. I do see NaNs when the cmor_coords fix is disabled, but I do not see any NaNs anymore when the cmor_coords fix is activated. This seems to me like these padding fixes are unnecessary. Can you confirm this, or otherwise give me the exact shapefile on which it doesn't work for you without the paddings? I would really like to get this merged today, to get it in with the release!

Reply:
@JaroCamphuijsen mate yes they are needed - please see the test for nan's, both the land and ocean NE files we have in the `ne_masks` dir need padding otherwise you end up with NaN strips; these files are ona -180-180 grid, so `cmor_coords` is `False` in this case, whereas this is needed for cmor-like coords too (0-360) I wouldn't know but better not risk it. Note that the check on cmor-like coords just rotates the lon to be on a 0-360 axis, no operation on the missing data strips. Yes, let's get this in today! :beer:

Reply:
@valeriupredoi 
I'm running the tests, but these also pass when I turn off the padding fixes (i.e. comment out line 538-541 in _area.py), so for the unit tests, this is not needed.

Reply:
yes, you are right - because they are masked and not NaN's - I got carried away by your previous comment (about NaN's) and plugged in the test, they are in fact masked strips

Reply:
ok fixed the test and simplified it now - you will see that if you set those pad params to False in the code that bit will be fully masked

Reply:
You are right, it was my sleep deprivation after all. I found the problem and see how your implementation fixes this, however I still think we should talk about this, since correcting for the quirks of the ne_shapefiles in the preprocessors of esmvalcore, doesn't seem right to me.

Reply:
yup! totally agree :beer: